orchestration/dap_orchestration/solids.py

```diff
@@ -0,0 +1,133 @@
+from dagster import Bool, String, solid, configured
+from dagster.core.execution.context.compute import AbstractComputeExecutionContext
+
+
+@solid(
+    required_resource_keys={"beam_runner", "refresh_directory"},
+    config_schema={
+        "pull_data_dictionaries": Bool,
+        "end_time": String,
+        "target_class": String,
+        "api_token": String,
+        "output_prefix": String,
+    }
+)
+def extract_records(context: AbstractComputeExecutionContext) -> None:
+    """
+    :return: Returns the path to extracted files.
+    """
+    arg_dict = {
+        "pullDataDictionaries": "true" if context.solid_config["pull_data_dictionaries"] else "false",
+        "outputPrefix": f"{context.resources.refresh_directory}/{context.solid_config['output_prefix']}",
+        "endTime": context.solid_config["end_time"],
+        "target_class": context.solid_config["target_class"],
+        "apiToken": context.solid_config["api_token"],
+        "scala_project": "dog-aging-hles-extraction"
+    }
+    context.resources.beam_runner.run(arg_dict)
+
+
+def _build_extract_config(config: dict[str, str], target_class: str, output_prefix: str) -> dict[str, str]:
```

Comment:
I'm not sure how useful an abstraction this is since we're just building a dictionary; do you think it'd be more readable to just return a dictionary in each of the configured solids below?

Reply:
I'm going to push back on this; the premise was to extract the thing that was varying between the 3 flavors of solids (the target_class) and centralize the rest of the config management. I think this is worth keeping for now.